### DISCUSSION TITLE
Support for parsing YAML

### DISCUSSION THREAD

**Discussion by PoisonousGame** [godotengine/godot-proposals#4151] [Engine Core]
https://www.geeksforgeeks.org/what-is-the-difference-between-yaml-and-json/#:~:text=YAML%3A%20It%20is%20a%20light,JSON%20with%20a%20YAML%20parser.

YAML: It is a light-weight, human-readable data-representation language. It is primarily designed to make the format easy to read while including complex features. Since YAML is a superset of JSON, it can parse JSON with a YAML parser.The extensions in YAML are .yaml or .yml. YAML specifications allow user-defined data types as well as explicit data typing.

<html>
<body>

YAML | JSON
-- | --
Comments are denoted with a hash/number sign. | Comments are not allowed.
Hierarchy is denoted by using double space characters. Tab characters are not allowed. | Objects and Arrays are denoted in braces and brackets.
String quotes are optional but it supports single and double quotes. | Strings must be in double quotes.
Root node can be any of the valid data types. | Root node must either be an array or an object.

</body>
</html>

**Comment by theraot** [comment]
I don't think this is something that needs to be in the core, nor something the core needs. So, an addon? It can be an addon. Somebody already did the addon: https://github.com/Beliaar/godot-yaml-asset Granted that it would need to be reworded for Godot 4. Does that work for you?

**Comment by Calinou** [comment]
There's a proposal tracking this: https://github.com/godotengine/godot-proposals/issues/941

However, as noted by @theraot, this is unlikely to be added in core. The built-in ConfigFile class already does a good job of reading and writing human-readable (and human-writable) configuration files. Only use JSON when you need its interoperability :slightly_smiling_face:
